python/ray/dag/compiled_dag_node.py

```diff
@@ -746,6 +743,15 @@ def _get_node_id(self):
             for idx in task.downstream_node_idxs:
                 frontier.append(idx)
 
+        from ray.dag.constants import RAY_ADAG_ENABLE_VERIFY_GRAPH
+
+        if RAY_ADAG_ENABLE_VERIFY_GRAPH:
```

Comment:
I think the verifier should always be on in the future. I use the environment variable as a feature flag to allow users to turn it off in case the verifier triggers a false alarm. Once we are confident enough in the verifier's accuracy, we can remove the feature flag. In that case, we may not need to expose this defensive feature flag in our public API (experimental_compile).